### PR TITLE
feat: support custom_header

### LINKED ISSUE #1: custom header in origin arguments?
We are  using Cloudfront as a reverse proxy for an Auth0 custom login domain. The configuration requires a custom header in the origin...
https://auth0.com/docs/custom-domains/configure-custom-domains-with-self-managed-certificates/configure-aws-cloudfront-for-use-as-reverse-proxy

is it possible to set up a customer header using this module?

### PR DESCRIPTION
**Pull request by karancode**
## Description
added support for `custom_header` option.

## Motivation and Context
Support for custom header is added. Fixes #1.

## Breaking Changes
N/A

## How Has This Been Tested?
Updated the example to add custom_header option as well. Validate the example & make plan for apply.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -124,3 +124,8 @@ variable "cache_behavior" {
   default     = null
 }
 
+variable "custom_header" {
```
Comment (antonbabenko) [inline variables.tf:127]: `custom_header` can be configured on individual origin's level, so it should be part of `var.origin` and not a separate variable.

Please update the code.

Comment (karancode) [reply variables.tf]: @antonbabenko thanks for the review. Sorry I missed that. 
Updated. PTAL.

**Comment by antonbabenko** [comment]
Thanks, @karancode! Remember that examples exist for very good reasons - they are 100% executable by anyone.

[v1.0.0](https://github.com/terraform-aws-modules/terraform-aws-cloudfront/releases/v1.0.0) has been just released.

**Comment by karancode** [comment]
Got it. Thank you @antonbabenko 🙇

**Pull request merged**